misc/msvs/props.template

```diff
@@ -18,6 +18,30 @@
   <ItemGroup Condition="%%CONDITION%%">
     %%EXTRA_ITEMS%%
   </ItemGroup>
+
+  <!-- Required for Rider below this line-->
+  <PropertyGroup Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <LocalDebuggerCommand Condition="'$(LocalDebuggerCommand)' == ''">$(NMakeOutput)</LocalDebuggerCommand>
+    <DefaultPlatformToolset>$(PlatformToolset)</DefaultPlatformToolset>
+  </PropertyGroup>
+
+  <Target Name="Build" Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <Exec Command="$(NMakeBuildCommandLine)"/>
+  </Target>
+  <Target Name="Rebuild" Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <Exec Command="$(NMakeReBuildCommandLine)"/>
+  </Target>
+  <Target Name="Clean" Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <Exec Command="$(NMakeCleanCommandLine)"/>
+  </Target>
+
+  <ItemDefinitionGroup Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <ClCompile>
+      <PreprocessorDefinitions>
+        $(NMakePreprocessorDefinitions)
+      </PreprocessorDefinitions>
+    </ClCompile>
+  </ItemDefinitionGroup>
```

Comment:
This is the same problem as the other comments below. Just because the platform toolset is clang, doesn't mean you're in Rider, or that the nmake support won't work. Adding these based on that assumption means that any clang-based builds in VS will run twice (if it doesn't break the builds entirely).

It's perfectly fine to generate more than one vcxproj and associated .generated.props files, one targeting Rider and another for VS, if you can't find a way to detect which IDE is running and what capabilities it has.

Reply:
It turns out I was confised a lot. But now it suddenly makes more sense. Thank you for pointers!